src/main/java/com/exasol/releasedroid/usecases/report/Result.java

```diff
@@ -1,21 +1,91 @@
 package com.exasol.releasedroid.usecases.report;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.exasol.releasedroid.usecases.request.PlatformName;
+
 /**
- * An abstract base for Release Droid actions results.
+ * A result of Release Droid action.
  */
-public abstract class Result {
+public class Result {
     private final boolean successful;
+    private final String actionDescription;
+    private final String message;
+    private final List<PlatformName> platformNames = new ArrayList<>();
 
-    protected Result(final boolean successful) {
+    protected Result(final boolean successful, final String actionDescription, final String message) {
         this.successful = successful;
+        this.actionDescription = actionDescription;
+        this.message = message;
     }
 
     /**
      * Check is a validation is successful.
      *
      * @return true if a validation is successful
      */
-    public boolean isSuccessful() {
-        return this.successful;
+    public boolean isFailed() {
+        return !this.successful;
+    }
+
+    /**
+     * Get the action description.
+     *
+     * @return action description
+     */
+    public String getActionDescription() {
+        return this.actionDescription;
+    }
+
+    /**
+     * Get the message.
+     *
+     * @return message
+     */
+    public String getMessage() {
+        return this.message;
+    }
+
+    /**
+     * Get the platform names.
+     *
+     * @return platform names
+     */
+    public List<PlatformName> getPlatformNames() {
+        return this.platformNames;
+    }
+
+    protected void addPlatforms(final List<PlatformName> platformNames) {
+        for (final PlatformName platformName : platformNames) {
+            if (!this.platformNames.contains(platformName)) {
```

Comment:
maybe you can use a `Set` for this

Reply:
The formatting part where I print the names is not stable in tests if I use Set. As we won't ever have more than 10 platforms here, I think it's not really a big performance issue to have an O(N^2) complexity here